SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestRPCTimeout/SpatialTestRPCTimeout.cpp

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SpatialTestRPCTimeout.h"
+#include "Components/BoxComponent.h"
+#include "GameFramework/Character.h"
+#include "GameFramework/PlayerController.h"
+#include "SpatialFunctionalTestFlowController.h"
+#include "SpatialTestRPCTimeoutGameMode.h"
+#include "SpatialTestRPCTimeoutPlayerController.h"
+
+/**
+ * This test ensures that RPC calls with unresolved parameters are queued until their parameters are correctly resolved.
+ * This test contains 1 Server and 2 Client workers running in multiple processes.
+ *
+ * The flow is as follows:
+ * - Setup:
+ *  - Launch at least one client in a separate process
+ * - Test:
+ *  - All clients launched outside of the editor process must ensure that the referenced material asset is not in memory at the start of the
+ * test.
+ *  - All clients launched outside of the editor process must successfully resolve the material asset before passing it into a client RPC
+ * function.
+ */
+
+ASpatialTestRPCTimeout::ASpatialTestRPCTimeout()
+	: Super()
+{
+	Author = "Iwan";
+	Description = TEXT(
+		"This test calls an RPC with an asset that was softly referenced and check that it will be asynchronously loaded with a timeout of "
+		"0.");
+}
+
+void ASpatialTestRPCTimeout::CreateCustomContentForMap()
+{
+	GetWorldSettingsForMap()->DefaultGameMode = ASpatialTestRPCTimeoutGameMode::StaticClass();
+}
+
+void ASpatialTestRPCTimeout::PrepareTest()
+{
+	Super::PrepareTest();
+
+	AddStep(
+		TEXT("Check that the material was not initially loaded on non-editor clients"), FWorkerDefinition::AllClients, nullptr, nullptr,
+		[this](float DeltaTime) {
+			if (!GIsEditor)
```

Comment:
Maybe worth adding a comment that the `(!GIsEditor)` indicates this client is the external process one. 
Perhaps we should make a testing helper to make this more obvious and increase user friendlyness?

Reply:
Yeah this was part of the step description (line 44) but it would be a good idea to make it more explicit (I can add an additional comment). 

I think the ideal solution for maximum user frendliness here is to create a new worker defintion `FWorkerDefinition::ExternalProcessClients`, although that may take some time to implement and test to make sure it works correctly

Reply:
A macro or helper function which just gives the call an explicit name will do fine for now. Even if it just is 
``` bool IsExternalProcessClient() {
	// The internal process client will have GIsEditor set to true, the external process clients have it false. This is because they don't share memory with the editor process.	
	return bIsClient && !GIsEditor;
}
```

Reply:
Ah alright I agree, I'll put that in that's plenty more readable

Still, having some generic worker categories like `FWorkerDefinition::ExternalProcessWorkers`, `FWorkerDefinition::ExternalProcessClients`, and `FWorkerDefinition::ExternalProcessServers` are probably something we would want to look into in future, would make the code a lot cleaner

Reply:
Change has been added!